watchtower-plugin/src/retrier.rs

```diff
@@ -650,15 +678,20 @@ mod tests {
             .unwrap()
             .add_pending_appointment(tower_id, &appointment);
 
+        #[cfg(not(feature = "accountable"))]
         // Prepare the mock response
         let mut add_appointment_receipt = AppointmentReceipt::new(
             cryptography::sign(&appointment.to_vec(), &wt_client.lock().unwrap().user_sk).unwrap(),
             42,
         );
+        #[cfg(not(feature = "accountable"))]
         add_appointment_receipt.sign(&tower_sk);
+        #[cfg(feature = "accountable")]
         let add_appointment_response =
             get_dummy_add_appointment_response(appointment.locator, &add_appointment_receipt);
 
+        #[cfg(not(feature = "accountable"))]
+        let add_appointment_response = get_dummy_add_appointment_response(appointment.locator);
```

Comment:
This is all over the place. You are signing when not accountable, which should not even be an allowed method. Re-check this test